prometheus/values.yaml

```diff
@@ -276,5 +314,83 @@ prometheus-postgres-exporter:
+  PrometheusRules: {}
+  PrometheusAlerts:
+    prometheus-main:
+      team: sre
+      selector:
+        prometheus: main
+      CriticalMetric:
+        interval: "31s"
+        rules:
+          # pvc
+          PVC_low_capacity:
+            for: "15m"
+            description: "Volume {{ $labels.persistentvolumeclaim }} located in namespace {{ $labels.namespace }} (on {{ $labels.node }}) has capacity < 5%"
+            expr: (kubelet_volume_stats_available_bytes / on(instance,namespace,node,persistentvolumeclaim,service) kubelet_volume_stats_capacity_bytes*100) < 5
+          # Postgres
+          Available_postgresql_connections_are_running_out:
+            for: "15m"
+            description: "There are {{ $value }} available postgres (pod {{ $labels.pod }}) connections left, located in namespace {{ $labels.namespace }}"
+            expr: (sum by (pod, namespace) (pg_stat_activity_count) / sum by (pod, namespace) (pg_settings_max_connections) > 0.90) or (sum by (pod, namespace) (pg_settings_max_connections) - sum by (pod, namespace) (pg_stat_activity_count) < 20)
+          # Resources
+          OOM_killed_pods:
+            for: "0m"
+            description: "Container {{ $labels.exported_container }} in pod {{ $labels.exported_pod }} located in namespace {{ $labels.exported_namespace }} (node {{ $labels.node }}) was killed by OOM killer"
+            expr: ((increase(kube_pod_container_status_restarts_total[10m]) > 0) and ignoring (reason) kube_pod_container_status_last_terminated_reason{reason="OOMKilled"}) * on (exported_pod) group_left (node) kube_pod_info > 0
+          # Deployment
+          Pods_waiting_state:
+            for: "20m"
+            description: "Container {{ $labels.container }} in {{ $labels.pod }} waiting with reason {{ $labels.reason }} for >20 minutes"
+            expr: sum by (pod, container, namespace, reason) (kube_pod_container_status_waiting_reason{exported_namespace!="kube-system", reason!="OOMKilled"})>0
+          Pods_waiting_state_kube-system:
+            for: "1h"
+            description: "Container {{ $labels.container }} in {{ $labels.pod }} waiting with reason {{ $labels.reason }} for >1h"
+            expr: sum by (pod, container, namespace, reason) (kube_pod_container_status_waiting_reason{exported_namespace="kube-system", reason!="OOMKilled"})>0
+          Pod_replicas_not_ready:
+            for: "15m"
+            description: "Pods replicas less than specified for '{{ $labels.deployment }}' for > 15m"
+            expr: (kube_deployment_spec_replicas * on (deployment) group_left (container) kube_deployment_labels) != (kube_deployment_status_replicas_available * on (deployment) group_left (container) kube_deployment_labels)
+          Daemonset_not_ready:
+            for: "15m"
+            description: "daemonset '{{ $labels.daemonset }}' not scheduled in 15m"
+            expr: kube_daemonset_status_number_misscheduled > 1
+          RabbitMQ_High_memory(watermark)_usage:
```

Comment:
What will happen if a company/project/env does not have Rabbitmq?

Reply:
The alert will not work